src/rulenode_operators.jl

```diff
@@ -1,7 +1,71 @@
-HerbCore.RuleNode(ind::Int, grammar::AbstractGrammar) = RuleNode(ind, nothing, [Hole(get_domain(grammar, type)) for type ∈ grammar.childtypes[ind]])
-HerbCore.RuleNode(ind::Int, _val::Any, grammar::AbstractGrammar) = RuleNode(ind, _val, [Hole(get_domain(grammar, type)) for type ∈ grammar.childtypes[ind]])
+"""
+    holes_from_child_types(index::Integer, grammar::ContextSensitiveGrammar)
+
+Given the `index` of a rule in a `grammar`, create a vector of [`Hole`](@ref)s
+corresponding to the children of the rule at `index`.
+"""
+function holes_from_child_types(index::Integer, grammar::ContextSensitiveGrammar)
```

Comment:
The name/purpose of this function could be clearer, as pointed out by @THinnerichs

Reply:
Dropping these functions instead as their utility is not that clear anyway, and we are not currently using them anywhere.